experimenter/experimenter/jetstream/models.py

```diff
@@ -329,71 +334,71 @@ def append_conversion_count(self, primary_metrics_set):
 """
 
 
+# XXX: does this need @lru_cache ?
 def create_results_object_model(data: JetstreamData):
-    branches = {}
-    metrics = {}
-
-    for jetstream_data_point in data:
-        branches[jetstream_data_point.branch] = {}
+    branches = [data_point.branch for data_point in data]
 
     # create a dynamic model with all branches, leveraging BranchComparisonData
-    branches_data = {b: BranchComparisonData() for b in branches}
     PairwiseBranchComparisonData = create_model(
         "PairwiseBranchComparisonData",
-        **branches_data,
+        **{
+            branch: (BranchComparisonData, Field(default_factory=BranchComparisonData))
```

Comment:
What's the advantage of declaring a `Field` with `default_factory` vs just setting it to an instance?

Reply:
I think pydantic will magically transform stuff like `x = F()` to `x = Field(default_factory=F)` if F isnt hashable but i think this is more clear as to whats going on.